### PR TITLE
Store the Sync wrapper in a variable for clean removal.

### PR DESCRIPTION
**Pull request by archisgore**
This commit primarily fixes this issue:
https://github.com/koalyptus/TableFilter/issues/258

The fix here is to store a Sync call closed over "this" straight within
the object we're referencing. This having access to that object, we can
always retrieve the exact function reference/pointer/instance for removal
from handling events.

### CODE REVIEW

File: src/modules/hash.js [hunk 1]
```diff
@@ -38,10 +38,11 @@ export class Hash {
         }
 
         this.lastHash = location.hash;
+        this.boundSync = this.sync.bind(this); //Store a bound sync wrapper for future use.
```
Comment (koalyptus) [inline src/modules/hash.js]: Linter is complaining about the line length, enforcing 80 chars in this project:
https://travis-ci.org/koalyptus/TableFilter/builds/144638612#L366
You can place the comment above the line if you want.

**Comment by koalyptus** [comment]
Thanks for your contribution @archisgore, I appreciate!
Can you please increment the package version to `0.2.43` in `package.json`:
https://github.com/koalyptus/TableFilter/blob/master/package.json#L3

Cheers

File: src/modules/hash.js [hunk 2]
```diff
@@ -90,8 +91,8 @@ export class Hash {
      */
     destroy() {
         this.emitter.off(['state-changed'], (tf, state) => this.update(state));
-        this.emitter.off(['initialized'], () => this.sync());
-        removeEvt(root, 'hashchange', () => this.sync());
+        this.emitter.off(['initialized'], () => this.boundSync);
+        removeEvt(root, 'hashchange', () => this.boundSync);
```
Comment (koalyptus) [inline src/modules/hash.js]: I think we don't need the arrow function here as per ln 44 and 45.

**Comment by archisgore** [comment]
Thanks for your feedback and quick reply. I made the changes and updated the code.

File: src/modules/hash.js [hunk 3]
```diff
@@ -38,10 +38,13 @@ export class Hash {
         }
 
         this.lastHash = location.hash;
+
```
Comment (koalyptus) [inline src/modules/hash.js]: Linter is now complaining of trailing white-spaces:
https://travis-ci.org/koalyptus/TableFilter/builds/144860663#L366

File: src/modules/hash.js [hunk 4]
```diff
@@ -38,10 +38,13 @@ export class Hash {
         }
 
         this.lastHash = location.hash;
+ 
+        //Store a bound sync wrapper for future use.
+        this.boundSync = this.sync.bind(this);
```
Comment (koalyptus) [inline src/modules/hash.js]: Same issue here:
https://travis-ci.org/koalyptus/TableFilter/builds/144860663#L367

**Comment by koalyptus** [comment]
Almost there :smile: 
I advise you to `npm install` and run the linter with the  `grunt eslint` command to avoid CI build to fail.
Also please bump the package version.
Thanks in advance

**Comment by koalyptus** [comment]
Tx for your patience:
:+1:

**Pull request merged**